### ISSUE TITLE
[SAFARI] vendor `-webkit-text-fill-color` not being applied to `color`

### ISSUE THREAD

**Issue by pedroapfilho** [styled-components/styled-components#2891]
## System:
	 - OS: macOS 10.15.1
	 - CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
	 - Memory: 160.35 MB / 16.00 GB
	 - Shell: 5.7.1 - /bin/zsh
	## Binaries:
	 - Node: 12.13.0 - /usr/local/bin/node
	 - Yarn: 1.19.1 - /usr/local/bin/yarn
	 - npm: 6.12.0 - /usr/local/bin/npm
	## npmPackages:
	 - styled-components: ^4.4.0 => 4.4.0

## Reproduction

```js
  const StyledInput = styled.input`
  ${props =>
    props.disabled &&
    css`
      && {
        border: solid 1px ${theme.palette.grey[200]};
        background-color: ${theme.palette.grey[200]};
        color: ${theme.palette.text.disabled};
        > .MuiSvgIcon-root {
          color: ${theme.palette.text.disabled};
        }
      }
    `}
  `
```

## Steps to reproduce

A bug was reported about the color of a disabled input on Safari, I got there to test and it seems that I had to add: `-webkit-text-fill-color: ${theme.palette.text.disabled};` to make it work. I thought it would be added by `styled-components` in runtime.

## Expected Behavior

![Screenshot 2019-11-22 at 10 47 34](https://user-images.githubusercontent.com/13142568/69420031-bd528e80-0d15-11ea-9278-d6bb4f04af5e.png)

## Actual Behavior

![Screenshot 2019-11-22 at 10 49 07](https://user-images.githubusercontent.com/13142568/69420047-c3486f80-0d15-11ea-9d4b-4b1c546d5c41.png)

**Comment by probablyup** [comment]
Hmm I've styled many safari inputs and never needed to use that vendor property. It might be better to just do normal `[disabled]` CSS:

```jsx
const StyledInput = styled.input`
  &[disabled] {
    border: solid 1px ${theme.palette.grey[200]};
    background-color: ${theme.palette.grey[200]};
    color: ${theme.palette.text.disabled};
    > .MuiSvgIcon-root {
      color: ${theme.palette.text.disabled};
    }
  }
`;

```

**Comment by Mustack** [comment]
I came across this issue because I was hoping styled-components could add this vendor property. I ran into a similar issue. The text color in Safari required the vendor prop because the element is inside of a button that has all `all: unset;`.

Here's a codepen that demonstrates the problem. https://codepen.io/mustack/pen/oNxbdWK
Here's a Stack Overflow that describes the issue: https://stackoverflow.com/questions/61019987/cant-set-text-color-of-all-unset-button-in-safari
Here's the Safari bug that SO issue links to: https://bugs.webkit.org/show_bug.cgi?id=158782

The reason I ran into this issue is that I have an element that is clickable and I didn't want to add an onClick on a div for accessibility reasons. I wrapped my component in a <button> to get proper tab-nav and more semantic markup but I didn't want all of the default browser styles on that button so I used `all: unset` and styled the contents myself.

I'm able to work around this by adding the vendor prop myself but I wanted to share my use-case here and see if you would be interested in adding this to styled-components.
